packages/core/test/plugins/use-error-handler.spec.ts

```diff
@@ -139,4 +141,43 @@ describe('useErrorHandler', () => {
       }),
     );
   });
+
+  it('should invoke error handler when error happens during incremental execution', async () => {
+    const schema = makeExecutableSchema({
+      typeDefs: /* GraphQL */ `
+        directive @defer on FRAGMENT_SPREAD | INLINE_FRAGMENT
+
+        type Query {
+          foo: String
+        }
+      `,
+      resolvers: {
+        Query: {
+          foo: () => {
+            throw new Error('kaboom');
+          },
+        },
+      },
+    });
+
+    const mockHandler = jest.fn();
+    const testInstance = createTestkit(
+      [
+        useEngine({ ...GraphQLJS, execute: normalizedExecutor, subscribe: normalizedExecutor }),
+        useErrorHandler(mockHandler),
+      ],
+      schema,
+    );
+    const result = (await testInstance.execute(`query { ... @defer { foo } }`)) as
+      | ExecutionResult
+      | IncrementalExecutionResult
+      | AsyncIterableIterator<ExecutionResult>
+      | AsyncIterableIterator<IncrementalExecutionResult>;
+    if ('next' in result) {
+      for await (const _r of result) {
+      }
+    }
```

Comment:
You can use some helper functions from the test package to help you handle stream result.

```suggestion
    const result = await testInstance.execute(`query { ... @defer { foo } }`)
    assertStreamExecutionValue(result); // Verify that the result is a stream
    await collectAsyncIteratorValues(result); // Wait for the end of the stream and returns all the incremental responses
```